### PR TITLE
doc(adding-a-cask): fix formatting, adding more hints

### PR DESCRIPTION
**Pull request by muescha**
changed documentation:

- add clear difference between `new-cask-name` and `new-cask-name-branch`
- format to see step by step commands better as bash code snippet
- add remove of `HOMEBREW_NO_AUTO_UPDATE` variable

### CODE REVIEW

File: doc/development/adding_a_cask.md [hunk 1]
```diff
@@ -251,16 +251,29 @@ $ git status
...
 
-Stage your Cask with `git add Casks/my-new-cask.rb`. You can view the changes that are to be committed with `git diff --cached`.
+Stage your Cask with: 
 
-Commit your changes with `git commit -v`.
+```bash
+$ git add Casks/my-new-cask.rb
+``` 
+
+You can view the changes that are to be committed with:
```
Comment (vitorgalvao) [inline doc/development/adding_a_cask.md]: Needs empty line below.

File: doc/development/adding_a_cask.md [hunk 2]
```diff
@@ -290,19 +303,38 @@ Examples of difficult, unclear commit summaries:
...
 
 ### Filing a Pull Request on GitHub
 
-Now go to the [`homebrew-cask` GitHub repository](https://github.com/Homebrew/homebrew-cask). GitHub will often show your `my-new-cask` branch with a handy button to `Compare & pull request`. Otherwise, click the `New pull request` button and choose to `compare across forks`. The base fork should be `Homebrew/homebrew-cask @ master`, and the head fork should be `my-github-username/homebrew-cask @ my-new-cask`. You can also add any further comments to your pull request at this stage.
+#### a) suggestion from git push
+
+The `git push` command prints a suggestion to create a pull request:
+
+```
+remote: Create a pull request for 'new-cask-guppy' on GitHub by visiting:
```
Comment (vitorgalvao) [inline doc/development/adding_a_cask.md]: `new-cask-guppy` to `my-new-cask`.

File: doc/development/adding_a_cask.md [hunk 3]
```diff
@@ -290,19 +303,38 @@ Examples of difficult, unclear commit summaries:
 
 ### Pushing
 
-Push your changes to your GitHub account:
+Push your changes from the branch `my-new-cask-branch` to your GitHub account:
 
 ```bash
-$ git push <my-github-username> my-new-cask
+$ git push <my-github-username> my-new-cask-branch
```
Comment (vitorgalvao) [inline doc/development/adding_a_cask.md]: Please change instances of `<var>` to `{{var}}`, for consistency with other documents.

File: doc/development/adding_a_cask.md [hunk 4]
```diff
@@ -290,19 +303,38 @@ Examples of difficult, unclear commit summaries:
...
 ### Filing a Pull Request on GitHub
 
-Now go to the [`homebrew-cask` GitHub repository](https://github.com/Homebrew/homebrew-cask). GitHub will often show your `my-new-cask` branch with a handy button to `Compare & pull request`. Otherwise, click the `New pull request` button and choose to `compare across forks`. The base fork should be `Homebrew/homebrew-cask @ master`, and the head fork should be `my-github-username/homebrew-cask @ my-new-cask`. You can also add any further comments to your pull request at this stage.
+#### a) suggestion from git push
+
+The `git push` command prints a suggestion to create a pull request:
+
+```
+remote: Create a pull request for 'new-cask-guppy' on GitHub by visiting:                                          
+remote:      https://github.com/<my-github-username>/homebrew-cask/pull/new/my-new-cask-branch
```
Comment (vitorgalvao) [inline doc/development/adding_a_cask.md]: `my-new-cask-branch` needs to be between `{{}}`.

File: doc/development/adding_a_cask.md [hunk 5]
```diff
@@ -312,3 +344,16 @@ After your Pull Request is submitted, you should get yourself back onto `master`
...
 git checkout master
 ```
+
+if you set the variable `HOMEBREW_NO_AUTO_UPDATE` then clean it up with:
+
+```bash
+unset HOMEBREW_NO_AUTO_UPDATE
+```
+
+or
```
Comment (vitorgalvao) [inline doc/development/adding_a_cask.md]: Remove this second set. Instructions should be as brief as possible.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]
Reviewed on a phone. Might not have caught everything.

Nice improvements. Thank you.

File: doc/development/adding_a_cask.md (hunk 1 above)
Comment (muescha) [reply doc/development/adding_a_cask.md]: ok

File: doc/development/adding_a_cask.md (hunk 3 above)
Comment (muescha) [reply doc/development/adding_a_cask.md]: ok

Comment (muescha) [reply doc/development/adding_a_cask.md]: should i also change this snippet?
```
shasum -a 256 <file>
```

File: doc/development/adding_a_cask.md (hunk 2 above)
Comment (muescha) [reply doc/development/adding_a_cask.md]: fixed

File: doc/development/adding_a_cask.md (hunk 4 above)
Comment (muescha) [reply doc/development/adding_a_cask.md]: fixed

File: doc/development/adding_a_cask.md (hunk 5 above)
Comment (muescha) [reply doc/development/adding_a_cask.md]: the first is documented anywhere but not work on my upterm ... but the second work, so i add it.

i will remove it

**Comment by muescha** [comment]
updated from comments

File: doc/development/adding_a_cask.md (hunk 3 above)
Comment (vitorgalvao) [reply doc/development/adding_a_cask.md]: Yes.

**Pull request merged**